x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/customprocessing/CustomProcessorFactory.java

```diff
@@ -21,7 +22,14 @@ public CustomProcessorFactory(List<String> fieldNames) {
 
     public CustomProcessor create(DataFrameAnalysis analysis) {
         if (analysis instanceof Regression) {
-            return new RegressionCustomProcessor(fieldNames, (Regression) analysis);
+            Regression regression = (Regression) analysis;
+            return new DatasetSplittingCustomProcessor(
+                fieldNames, regression.getDependentVariable(), regression.getTrainingPercent());
+        }
+        if (analysis instanceof Classification) {
+            Classification classification = (Classification) analysis;
+            return new DatasetSplittingCustomProcessor(
+                fieldNames, classification.getDependentVariable(), classification.getTrainingPercent());
```

Comment:
It almost seems like we need a new interface for the different analysis. 

Unsupervised vs supervised... But that might be a future refactoring

Reply:
> It almost seems like we need a new interface for the different analysis.

Yes, we may end up doing that.

> But that might be a future refactoring

Agree, let's not add more interfaces too early.